Compiler error error: use 'template' keyword to treat 'connect' as a dependent template name in BounceBackNVE

Internal plugin v0.9.1 (commit 0c88936a)  (hoomd v2.6.0) compiled on mac with Apple clang version 11.0.0 (clang-1100.0.33.17, no cuda) results in

```
/Users/statt/Programs/hoomd-blue/hoomd/azplugins/BounceBackNVE.h:91:35: error: use 'template' keyword to treat 'connect' as a dependent template name
    m_pdata->getBoxChangeSignal().connect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);


/Users/statt/Programs/hoomd-blue/hoomd/azplugins/BounceBackNVE.h:99:35: error: use 'template' keyword to treat 'disconnect' as a dependent template name
    m_pdata->getBoxChangeSignal().disconnect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);
```

external plugin compiles but crashes at runtime with
```
  File "hist_class_test.hoomd", line 9, in <module>
    import azplugins
  File "/Users/statt/Programs/azplugins-0.7.0/azplugins/__init__.py", line 13, in <module>
    from . import _azplugins
ImportError: dlopen(/Users/statt/Programs/azplugins-0.7.0/azplugins/_azplugins.cpython-37m-darwin.so, 2): Symbol not found: __ZN24IntegrationMethodTwoStep17getRotationalNDOFESt10shared_ptrI13ParticleGroupE
  Referenced from: /Users/statt/Programs/azplugins-0.7.0/azplugins/_azplugins.cpython-37m-darwin.so
  Expected in: flat namespace
 in /Users/statt/Programs/azplugins-0.7.0/azplugins/_azplugins.cpython-37m-darwin.so
```

will test if this is mac specific, user error or something else.

--- a/azplugins/BounceBackNVE.h
+++ b/azplugins/BounceBackNVE.h
@@ -88,7 +88,7 @@ BounceBackNVE<Geometry>::BounceBackNVE(std::shared_ptr<SystemDefinition> sysdef,
     {
     m_exec_conf->msg->notice(5) << "Constructing BounceBackNVE + " << Geometry::getName() << std::endl;
 
-    m_pdata->getBoxChangeSignal().connect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);
+    m_pdata->getBoxChangeSignal().template connect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);
     }
 
 template<class Geometry>
@@ -96,7 +96,7 @@ BounceBackNVE<Geometry>::~BounceBackNVE()
     {
     m_exec_conf->msg->notice(5) << "Destroying BounceBackNVE + " << Geometry::getName() << std::endl;
 
-    m_pdata->getBoxChangeSignal().disconnect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);
+    m_pdata->getBoxChangeSignal().template disconnect<BounceBackNVE<Geometry>, &BounceBackNVE<Geometry>::requestValidate>(this);
     }
 
 template<class Geometry>

